lib/android/src/main/java/com/airbnb/android/react/maps/AirMapMbTile.java

```diff
@@ -0,0 +1,139 @@
+package com.airbnb.android.react.maps;
+
+import android.content.Context;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.Tile;
+import com.google.android.gms.maps.model.TileOverlay;
+import com.google.android.gms.maps.model.TileOverlayOptions;
+import com.google.android.gms.maps.model.TileProvider;
+
+import java.io.File;
+
+import android.os.Environment;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.Cursor;
+
+/**
+ * Created by Christoph Lambio on 30/03/2018.
+ * Based on AirMapLocalTileManager.java
+ * Copyright (c) zavadpe
+ */
+
+public class AirMapMbTile extends AirMapFeature {
+
+    class AIRMapMbTileProvider implements TileProvider {
+        private static final int BUFFER_SIZE = 16 * 1024;
+        private int tileSize;
+        private String pathTemplate;
+
+
+        public AIRMapMbTileProvider(int tileSizet, String pathTemplate) {
+            this.tileSize = tileSizet;
+            this.pathTemplate = pathTemplate;
+        }
+
+        @Override
+        public Tile getTile(int x, int y, int zoom) {
+            byte[] image = readTileImage(x, y, zoom);
+            return image == null ? TileProvider.NO_TILE : new Tile(this.tileSize, this.tileSize, image);
+        }
+
+        public void setPathTemplate(String pathTemplate) {
+            this.pathTemplate = pathTemplate;
+        }
+
+        public void setTileSize(int tileSize) {
+            this.tileSize = tileSize;
+        }
+
+        private byte[] readTileImage(int x, int y, int zoom) {
+            String rawQuery = "SELECT * FROM map INNER JOIN images ON map.tile_id = images.tile_id WHERE map.zoom_level = {z} AND map.tile_column = {x} AND map.tile_row = {y}";
+
+            try {
+                SQLiteDatabase offlineDataDatabase = SQLiteDatabase.openDatabase(this.pathTemplate, null, SQLiteDatabase.OPEN_READONLY);
+                String query = rawQuery.replace("{x}", Integer.toString(x))
+                        .replace("{y}", Integer.toString(y))
+                        .replace("{z}", Integer.toString(zoom));
+                Cursor cursor = offlineDataDatabase.rawQuery(query, null);
+                if(cursor.moveToFirst()){
+                    byte[] tile = cursor.getBlob(5);
+                    cursor.close();
+                    offlineDataDatabase.close();
+                    return tile;
+                }
+                offlineDataDatabase.close();
+                return null;
+            } catch (Exception e) {
```

Comment:
Out of curiosity, what appears for the tile in this case? Is it blank? I see https://github.com/react-community/react-native-maps/blob/f1126a4f307084005f7b62b0dc50fbfcfc810cd4/lib/android/src/main/java/com/airbnb/android/react/maps/AirMapLocalTile.java#L62 does it too so I assume it's standard.

Reply:
Yep, should be blank.